File: 942. DI String Match.js
var diStringMatch = function(S) {
    let increase = 0;
    let length = S.length;
    let decrease = length;
    
    let result = [];
    
    for(let index = 0; index <= length; index++){
        let indicator = S.charAt(index);
        let num = (indicator == "D") ? decrease-- : increase++;
        result[index] = num;
    }
    
    return result;
};

File: javascript/942. DI String Match.java
class Solution {
    int[] result;
    public int[] diStringMatch(String S) {
        int length = S.length();
        result = new int[length+1];
        int increase = 0;
        int decrease = length;
        
        for(int index = 0; index < length; index++){
            int num = (S.charAt(index) == 'D') ? decrease-- : increase++;
            result[index] = num;
        }
        result[length] = increase;
        return result;
    }
}
